### PR TITLE
boat-angular update ngVersion to use >= range in dependencies

### PR DESCRIPTION
**Pull request by brianvabb**
Change:

Specify boat-angular ngVersion via range `>={{ngVersion}}` to support Angular version greater than range.

### CODE REVIEW

**Review by ngohungphuc — APPROVED** [review]

**Comment by santam85** [comment]
@brianvabb There seems to be some build error:
` Must use import to load ES Module: /home/runner/work/backbase-openapi-tools/backbase-openapi-tools/boat-maven-plugin/target/it/example/boat-generate/angular/target/http-module/node_modules/@angular/compiler-cli/bundles/index.js
[INFO] [INFO] require() of ES modules is not supported.
[INFO] [INFO] require() of /home/runner/work/backbase-openapi-tools/backbase-openapi-tools/boat-maven-plugin/target/it/example/boat-generate/angular/target/http-module/node_modules/@angular/compiler-cli/bundles/index.js from /home/runner/work/backbase-openapi-tools/backbase-openapi-tools/boat-maven-plugin/target/it/example/boat-generate/angular/target/http-module/node_modules/ng-packagr/lib/ts/tsconfig.js is an ES module file as it is a .js file whose nearest parent package.json contains "type": "module" which defines all .js files in that package scope as ES modules.`

Please have a look

**Comment by brianvabb** [comment]
https://github.com/Backbase/backbase-openapi-tools/pull/273#issuecomment-1170992637

It seems the build failure comes from a CommonJS module [trying to require()](https://www.the-guild.dev/blog/support-nodejs-esm) an ESM module. Is there some way to bump the Node version being used to build this project? Maybe a newer node version will handle transpiling to ESM modules.

**Comment by santam85** [comment]
> [#273 (comment)](https://github.com/Backbase/backbase-openapi-tools/pull/273#issuecomment-1170992637)
> 
> It seems the build failure comes from a CommonJS module [trying to require()](https://www.the-guild.dev/blog/support-nodejs-esm) an ESM module. Is there some way to bump the Node version being used to build this project? Maybe a newer node version will handle transpiling to ESM modules.

It should be possible to update it here: https://github.com/Backbase/backbase-openapi-tools/blob/5efaea2e3c2db5edafc222f8c276363c6f036292/boat-maven-plugin/src/it/example/boat-generate/angular/pom.xml#L67

File: boat-scaffold/src/main/templates/boat-angular/package.mustache [hunk 1]
```diff
@@ -15,18 +15,18 @@
     "build": "ng-packagr -p ng-package.json"
   },
   "peerDependencies": {
-    "@angular/core": "^{{ngVersion}}",
+    "@angular/core": ">={{ngVersion}}",
     {{#withMocks}}
     "@backbase/foundation-ang": "^{{foundationVersion}}",
     {{/withMocks}}
     "rxjs": "^{{rxjsVersion}}"
```
Comment (daiscog) [inline boat-scaffold/src/main/templates/boat-angular/package.mustache:22]: Do we want to do the same with rxjs and foundation-ang?

Comment (ngohungphuc) [reply boat-scaffold/src/main/templates/boat-angular/package.mustache:22]: @daiscog we want to support rxjs 6 and 7 too right ?

**Comment by ngohungphuc** [comment]
We can refer to this page for compatible version https://gist.github.com/LayZeeDK/c822cc812f75bb07b7c55d07ba2719b3

**Comment by ngohungphuc** [comment]
@brianvabb @santam85 I did some dig in the code and I see that we have failed test here output was false instead of true

     ```
   File output = new File("src/test/resources/raml-examples/converted/openapi.yaml");

        assertTrue(output.exists()); 
```

**Comment by bartveenstra** [comment]
On it

**Comment by ngohungphuc** [comment]
hi @bartveenstra thank you for your help but we still have the errors. Can you run this command inside boat-terminal folder ?
```shell
mvn -B verify sonar:sonar -X
```

**Comment by ngohungphuc** [comment]
So it's kind weird I run the command in the root I got the build failed on another project.
```shell
mvn clean install -DskipTests
```
Schermafbeelding 2022-07-01 om 13 49 56

**Comment by ngohungphuc** [comment]
@santam85 @brianvabb I think the problem come from the tsconfig file. We need to change to otherwise build will fail but i think this file is auto generated
```
        "target": "es2015",
        "module": "es2020",
```
Schermafbeelding 2022-07-01 om 13 59 33

**Comment by brianvabb** [comment]
> > [#273 (comment)](https://github.com/Backbase/backbase-openapi-tools/pull/273#issuecomment-1170992637)
> > It seems the build failure comes from a CommonJS module [trying to require()](https://www.the-guild.dev/blog/support-nodejs-esm) an ESM module. Is there some way to bump the Node version being used to build this project? Maybe a newer node version will handle transpiling to ESM modules.
> 
> It should be possible to update it here:
> 
> https://github.com/Backbase/backbase-openapi-tools/blob/5efaea2e3c2db5edafc222f8c276363c6f036292/boat-maven-plugin/src/it/example/boat-generate/angular/pom.xml#L67

Bumping node to `14.182` doesn't resolve that build error in sonar check, so we can rule that out as the fix - 
https://github.com/Backbase/backbase-openapi-tools/pull/273/commits/4e150210517da8cbafb4c2a8ddaddaf6c343e3e4

**Comment by santam85** [comment]
Not an ideal solution, as it's potentially unstable when new Angular releases come out. We should ideally look into generating a package.lock too, or having more complex defaults based on Angular versions. Works for now though.

**Comment by santam85** [comment]
> @santam85 I will merge this PR if we don't any problems

Just want to get one last review from @tcorral and @daiscog before

**Review by daiscog — APPROVED** [review]

**Pull request merged**